test/parallel/test-fetch-mock.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+require('../common');
+const { mock, test } = require('node:test');
+const assert = require('node:assert');
+
+test('should correctly stub globalThis.fetch', async () => {
+  const customFetch = async (url) => {
+    return {
+      text: async () => 'foo',
+    };
+  };
+
+  mock.method(globalThis, 'fetch', customFetch);
+
+  const response = await globalThis.fetch('some-url');
+  const text = await response.text();
+
+  assert.strictEqual(text, 'foo');
```

Comment:
```suggestion
  assert.strictEqual(text, 'foo');
  mock.restoreAll();
```

this should fix the failure.

Reply:
Thank you!!!